### PR TITLE
Fix `[tribe_events]` URL when using the month pagination and having p…

### PR DESCRIPTION
**Pull request by juanfra**
https://central.tri.be/issues/95720

Fix pre-existent issue with the URLs when using the shortcode and plain permalinks.

### CODE REVIEW

**Comment by juanfra** [comment]
I have to make some improvements. Please don't review the code right now.

**Comment by mitogh** [comment]
Adding the `hold` tag to the PR because of this comment. 

> I have to make some improvements. Please don't review the code right now.

**Comment by juanfra** [comment]
@mitogh Thanks! changes submitted, code ready to be checked.

File: src/resources/js/tribe-events-ajax-calendar.js [hunk 1]
```diff
@@ -301,6 +301,15 @@
 					url = td.cur_url.split("?")[0];
 				}
 
+				// if using the shortcode
+				if ( $( document.getElementById( 'tribe-events' ) ).is( '.tribe-events-shortcode' ) ) {
```
Comment (mitogh) [inline src/resources/js/tribe-events-ajax-calendar.js]: I think there is a `$wrapper` variable that holds the same reference to this DOM node so I don't think there's a need to have a new one.

File: src/resources/js/tribe-events-ajax-calendar.js [hunk 2]
```diff
@@ -429,12 +438,16 @@
...
 
+				// when having plain permalinks
 				if ( td.default_permalinks ) {
-					if( !ts.url_params.hasOwnProperty( 'post_type' ) ){
-						ts.url_params['post_type'] = config.events_post_type;
-					}
-					if( !ts.url_params.hasOwnProperty( 'eventDisplay' ) ){
-						ts.url_params['eventDisplay'] = ts.view;
+					// when not using the shorcode
+					if ( ! $( document.getElementById( 'tribe-events' ) ).is( '.tribe-events-shortcode' ) ) {
```
Comment (mitogh) [inline src/resources/js/tribe-events-ajax-calendar.js]: Same as noted above. 

Might be useful to have a function like: `is_event_shortcode()` that you can use here and above to keep things DRY

File: src/resources/js/tribe-events-ajax-calendar.js [hunk 3]
```diff
@@ -516,11 +529,14 @@
 								$( '#tribe-events.tribe-events-shortcode' ).length
 								|| ts.do_string
 						) {
-							if ( -1 !== td.cur_url.indexOf( '?' ) ) {
-								td.cur_url = td.cur_url.split( '?' )[0];
+							if ( ! td.default_permalinks ) {
```
Comment (mitogh) [inline src/resources/js/tribe-events-ajax-calendar.js]: On this one I would flip the  conditional and code just to have an easier reading of the code like: 

```js
if ( td.default_permalinks ) {
  td.cur_url = td.cur_url + '&' + ts.url_params;
} else {
  // ....
}
```

**Comment by juanfra** [comment]
Thanks @mitogh! I've just added these enhancements

**Pull request merged**